acceptance/tests/tasks/task_implementations.rb

```diff
@@ -0,0 +1,65 @@
+require 'pxp-agent/task_helper.rb'
+require 'pxp-agent/config_helper.rb'
+require 'puppet/acceptance/environment_utils.rb'
+
+test_name 'task downloads correct implementation' do
+
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  PUPPETSERVER_CONFIG_FILE = windows?(master) ?
+    '/cygdrive/c/ProgramData/PuppetLabs/puppetserver/etc/conf.d/webserver.conf'
+  : '/etc/puppetlabs/puppetserver/conf.d/webserver.conf'
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_hocon_using_puppet_certs(master, agent))
+      on agent, puppet('resource service pxp-agent ensure=running')
+
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  filename = 'init_rb.rb'
+  step 'Create the downloadable tasks on the master' do
+    # Make the static content path
+    env_name = File.basename(__FILE__, '.*')
+    @static_content_path = File.join(environmentpath, mk_tmp_environment(env_name))
+
+    # Create the task
+    task_body = "#!/opt/puppetlabs/puppet/bin/ruby\nputs ENV['PT_message']"
+    taskpath = "#{@static_content_path}/#{filename}"
+    create_remote_file(master, taskpath, task_body)
+    on master, "chmod 1777 #{taskpath}"
+    @sha256 = Digest::SHA256.hexdigest(task_body + "\n")
```

Comment:
Where does the `\n` come from in the task body?

Reply:
I think that's automatically created on the target with `create_remote_file`. This is just copied around at this point.

Reply:
:+1: